plugins/inputs/tail/tail.go

```diff
@@ -217,72 +217,76 @@ func (t *Tail) tailNewFiles(fromBeginning bool) error {
 		if err != nil {
 			t.Log.Errorf("Glob %q failed to compile: %s", filepath, err.Error())
 		}
-		for _, file := range g.Match() {
-			if _, ok := t.tailers[file]; ok {
-				// we're already tailing this file
-				continue
-			}
 
-			var seek *tail.SeekInfo
-			if !t.Pipe && !fromBeginning {
-				if offset, ok := t.offsets[file]; ok {
-					t.Log.Debugf("Using offset %d for %q", offset, file)
-					seek = &tail.SeekInfo{
-						Whence: 0,
-						Offset: offset,
-					}
-				} else {
-					seek = &tail.SeekInfo{
-						Whence: 2,
-						Offset: 0,
-					}
-				}
-			}
+		// create a goroutine for each "tailer"
+                 t.wg.Add(1)
 
-			tailer, err := tail.TailFile(file,
-				tail.Config{
-					ReOpen:    true,
-					Follow:    true,
-					Location:  seek,
-					MustExist: true,
-					Poll:      poll,
-					Pipe:      t.Pipe,
-					Logger:    tail.DiscardingLogger,
-					OpenReaderFunc: func(rd io.Reader) io.Reader {
-						r, _ := utfbom.Skip(t.decoder.Reader(rd))
-						return r
-					},
-				})
-
-			if err != nil {
-				t.Log.Debugf("Failed to open file (%s): %v", file, err)
-				continue
-			}
+                 go func() {
+			defer t.wg.Done()
+		        for _, file := range g.Match() {
```

Comment:
Tis will not work if an input plugin instance specifies more than one file to tail. It shoudl be a separate thread per file, not per input plugin instance. Looks like `go func` should be inside this loop.

Reply:
> Tis will not work if an input plugin instance specifies more than one file to tail. It shoudl be a separate thread per file, not per input plugin instance. Looks like `go func` should be inside this loop.

OK, I will check this.